spec/system/candidate_interface/references/reference_selection/candidate_can_continuously_request_references_spec.rb

```diff
@@ -0,0 +1,195 @@
+require 'rails_helper'
+
+RSpec.feature 'References' do
+  include CandidateHelper
+
+  scenario 'the candidate can continue to request and add references on an unsubmitted application' do
+    given_i_am_signed_in
+    and_i_have_provided_my_personal_details
+    and_the_select_references_flag_is_active
+    and_i_have_three_reference_requests_pending
+
+    when_i_receive_two_references
+    then_i_still_have_a_reference_request_outstanding
+    and_i_can_add_more_reference_requests
+    and_i_can_receive_more_references
+  end
+
+  def given_i_am_signed_in
+    @candidate = create(:candidate)
+    login_as(@candidate)
+    @application = @candidate.current_application
+  end
+
+  def and_i_have_provided_my_personal_details
+    @candidate.current_application.update!(first_name: 'Mr', last_name: 'Bot')
+  end
+
+  def and_the_select_references_flag_is_active
+    FeatureFlag.activate(:reference_selection)
+  end
+
+  def and_i_have_three_reference_requests_pending
+    candidate_provides_three_referees
```

Comment:
Rather than manually filling these out we can just do

```
3.times { create(:reference, :feedback_provided, application_form: application_form) } 
```

since we already test that you can add more than 2 applications elsewhere